From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
Date: Tue, 13 May 2025 15:53:41 +0000
Subject: [PATCH] Search engine data patches to include Mullvad Leta to list of
 search engines

These changes allow inclusion of Mullvad Leta search engine to offered
list of search engines on default region provided by the OS and the
browser.
---
 ...llvad-Leta-to-list-of-search-engines.patch | 29 +++++++++++++++++++
 ...ent-maximum-number-of-search-engines.patch | 22 ++++++++++++++
 ...ment-version-for-search-engines-data.patch | 22 ++++++++++++++
 ...-to-list-of-search-engine-on-US-regi.patch | 22 ++++++++++++++
 4 files changed, 95 insertions(+)
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0001-Add-Mullvad-Leta-to-list-of-search-engines.patch
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0002-Increment-maximum-number-of-search-engines.patch
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0003-Increment-version-for-search-engines-data.patch
 create mode 100644 vanadium/subprojects_patches/third_party/search_engines_data/resources/0004-Add-Mullvad-Leta-to-list-of-search-engine-on-US-regi.patch

diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0001-Add-Mullvad-Leta-to-list-of-search-engines.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0001-Add-Mullvad-Leta-to-list-of-search-engines.patch
new file mode 100644
index 0000000000000..1c4af09eded98
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0001-Add-Mullvad-Leta-to-list-of-search-engines.patch
@@ -0,0 +1,29 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Tue, 13 May 2025 14:41:50 +0000
+Subject: [PATCH] Add Mullvad Leta to list of search engines
+
+---
+ definitions/prepopulated_engines.json | 9 +++++++++
+ 1 file changed, 9 insertions(+)
+
+diff --git a/definitions/prepopulated_engines.json b/definitions/prepopulated_engines.json
+index 931ed5f..6b63989 100644
+--- a/definitions/prepopulated_engines.json
++++ b/definitions/prepopulated_engines.json
+@@ -274,6 +274,15 @@
+       "id": 103
+     },
+ 
++    "mullvad_leta": {
++      "name": "Mullvad Leta",
++      "keyword": "leta.mullvad.net",
++      "favicon_url": "https://leta.mullvad.net/favicon.ico",
++      "search_url": "https://leta.mullvad.net/search?q={searchTerms}",
++      "type": "SEARCH_ENGINE_MULLVAD_LETA",
++      "id": 116
++    },
++
+     "naver": {
+       "name": "\ub124\uc774\ubc84",
+       "keyword": "naver.com",
diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0002-Increment-maximum-number-of-search-engines.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0002-Increment-maximum-number-of-search-engines.patch
new file mode 100644
index 0000000000000..c7977bd583aa4
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0002-Increment-maximum-number-of-search-engines.patch
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Tue, 13 May 2025 14:41:06 +0000
+Subject: [PATCH] Increment maximum number of search engines
+
+---
+ definitions/prepopulated_engines.json | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/definitions/prepopulated_engines.json b/definitions/prepopulated_engines.json
+index 6b63989..6ab6799 100644
+--- a/definitions/prepopulated_engines.json
++++ b/definitions/prepopulated_engines.json
+@@ -22,7 +22,7 @@
+     // use a lower number. This number is used by UMA stats and must be updated
+     // to include the full range or else not all engines will be counted in
+     // stats.
+-    "kMaxPrepopulatedEngineID": 115,
++    "kMaxPrepopulatedEngineID": 116,
+ 
+     // Increment this if you change the data in ways that mean users with
+     // existing data should get a new version. Otherwise, existing data may
diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0003-Increment-version-for-search-engines-data.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0003-Increment-version-for-search-engines-data.patch
new file mode 100644
index 0000000000000..128d8b4c26c37
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0003-Increment-version-for-search-engines-data.patch
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Tue, 13 May 2025 14:41:30 +0000
+Subject: [PATCH] Increment version for search engines data
+
+---
+ definitions/prepopulated_engines.json | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/definitions/prepopulated_engines.json b/definitions/prepopulated_engines.json
+index 6ab6799..f6effbf 100644
+--- a/definitions/prepopulated_engines.json
++++ b/definitions/prepopulated_engines.json
+@@ -28,7 +28,7 @@
+     // existing data should get a new version. Otherwise, existing data may
+     // continue to be used and updates made here will not always appear.
+     // Also then run tools/search_engine_choice/generate_search_engine_icons.py.
+-    "kCurrentDataVersion": 188
++    "kCurrentDataVersion": 189
+   },
+ 
+   // The following engines are included in country lists and are added to the
diff --git a/vanadium/subprojects_patches/third_party/search_engines_data/resources/0004-Add-Mullvad-Leta-to-list-of-search-engine-on-US-regi.patch b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0004-Add-Mullvad-Leta-to-list-of-search-engine-on-US-regi.patch
new file mode 100644
index 0000000000000..1eb8d602d203e
--- /dev/null
+++ b/vanadium/subprojects_patches/third_party/search_engines_data/resources/0004-Add-Mullvad-Leta-to-list-of-search-engine-on-US-regi.patch
@@ -0,0 +1,22 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: quh4gko8 <88831734+quh4gko8@users.noreply.github.com>
+Date: Tue, 13 May 2025 14:42:13 +0000
+Subject: [PATCH] Add Mullvad Leta to list of search engine on US region
+
+Default region based on OS and Vanadium settings is US.
+---
+ definitions/regional_settings.json | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/definitions/regional_settings.json b/definitions/regional_settings.json
+index ca7dc52..6753408 100644
+--- a/definitions/regional_settings.json
++++ b/definitions/regional_settings.json
+@@ -1117,6 +1117,7 @@
+         "&bing",
+         "&yahoo",
+         "&duckduckgo",
++        "&mullvad_leta",
+         "&yandex_com"
+       ]
+     },
